LeadCount.cls

```diff
@@ -0,0 +1,23 @@
+/*Write a SOQL query to find total number of Lead records by status by Lead Source. Store this 
+information in map and display the same.*/
+
+public class LeadCount{
+	public static void getLeadCount(){
+		Map<String,Integer> myMap = new Map<String,Integer>();
+		List<aggregateResult> results = [Select LeadSource,count(Status) total from Lead group by LeadSource];
+		String key;
+		Integer val;
+		for (AggregateResult ar : results){
+			key=(String)ar.get('LeadSource');
+			val=(Integer)ar.get('total');
+			if(!myMap.containsKey(key)){

```

Comment:
Does it really need to have contains check here? as group by will always return you grouped data for LeadSource and here your map key is also LeadSource. Then it seems like it is unnecessary check.